python-avd/pyavd/_schema/models/avd_list.py

```diff
@@ -98,20 +98,43 @@ def get(self, index: int, default: T | UndefinedType = Undefined) -> T_ItemType
     def append(self, item: T_ItemType) -> None:
         self._items.append(item)
 
+    def append_unique(self, item: T_ItemType) -> None:
+        """Append the item if not there already. Otherwise ignore."""
+        if item not in self._items:
+            self._items.append(item)
+
+    if TYPE_CHECKING:
+        append_new: type[T_ItemType]
+    else:
+
+        def append_new(self, *args: Any, **kwargs: Any) -> T_ItemType:
+            """Create a new instance with the given arguments and append to the list. Returns the new item."""
+            new_item = self._item_type(*args, **kwargs)
+            self.append(new_item)
+            return new_item
+
     def extend(self, items: Iterable[T_ItemType]) -> None:
         self._items.extend(items)
 
-    def _as_list(self, include_default_values: bool = False, strip_values: tuple = (None, [], {})) -> list:
+    def _strip_empties(self) -> None:
+        """In-place update the instance to remove data matching the given strip_values."""
+        if issubclass(self._item_type, AvdBase):
+            [item._strip_empties() for item in self._items]
+            self._items = [item for item in self._items if item]
```

Comment:
This makes for one less list creation:
(verified in Python 3.10 and 3.13)
```suggestion
            self._items = [striped_item for item in self._items if (striped_items := item._strip_empties())]
```

Reply:
Are you sure it is working? Remember strip_empties is not returning anything. It is doing inplace updates.

Reply:
good point - my scratch pad testing was working but I had a return.
Though we should not create the list line 122 then

```
https://github.com/aristanetworks/avd/pull/4762

Reply:
How can I use map when it is a method of the item itself? I would need lambda. I doubt that will be more efficient.

Reply:
I think there would be options but lets not waste more time on this. It gave for nice readings .. :)

Reply:
As discussed elsewhere we will stick with list comprehension.